frontend/src/ui/busy/busy.tsx

```diff
@@ -10,4 +10,28 @@ export const BusyOverlay = () => (
     <div className="busy-overlay">
         <Spinner/>
     </div>
-);
\ No newline at end of file
+);
+
+export const BusyRender = <T extends {}>(props: {promise: Promise<T>, children: (data: T) => Array<ReactElement> | ReactElement}) => {
+    const {promise} = props;
+    const [state, setState] = useState<{loading: boolean, data: T}>({loading: true, data: null});
+
+    useEffect(() => {
+        promise.then((data) => {
+            setState({
+                loading: false,
+                data
+            })
+        });
+    }, []);
+
+    if(state.loading){
+        return <BusyOverlay/>
+    }
+
+    return (
+        <>
+            {props.children(state.data)}
+        </>
+    )
```

Comment:
`return children(state.data);`

Reply:
Here I couldn't get rid of the fragment. I'm having an annoying ts error which I think is related to  the [generic not extending key, id ](https://stackoverflow.com/questions/55133907/react-with-typescript-type-is-missing-the-following-properties-from-type) 🤔

Reply:
have you changed the retun type of children to be `children: (...) => ReactNode` ?

Reply:
yes 😫